src/Statement/StatementList.php

```diff
@@ -282,4 +282,24 @@ public function isEmpty() {
 		return empty( $this->statements );
 	}
 
+	/**
+	 * @since 3.0
+	 *
+	 * @param Statement $statement
+	 *
+	 * @return int|bool
+	 */
+	public function indexOf( Statement $statement ) {
+		$index = 0;
+
+		foreach ( $this->statements as $s ) {
+			if ( $s->equals( $statement ) ) {
```

Comment:
Now this will do odd things if the list contains statements with no GUID. Dang, I don't know what's the right thing to do. `indexOfGuid( string $guid )`?

I don't think looking at use cases in Wikibase.git helps. Methods in this component should be self-explaining, not depend on use cases specific for a product.